index.html

```diff
@@ -847,90 +847,77 @@ <h2>Examples</h2>
-                if (this.hasOwnProperty(propertyName))
-                  resolve(this[propertyName]);
-                else
-                  reject('Property not found');
-              },
-              e => {
-                console.log("Error");
-              });
-          });
+          thing.setActionHandler( () => {
+            console.log("Resetting maximum");
+            thing.writeProperty("max", 0.0);
+          }, "reset");
+
+          thing.start();
+
+          setInterval( async () => {
+            let mock = Math.random()*100;
+            thing.writeProperty("temperature", mock);
+            let old = await thing.readProperty("max");
+            if (old < mock) {
+              thing.writeProperty("max", mock);
+              thing.emitEvent("onchange");
+            }
+          }, 1000);
+
         } catch (err) {
-           console.log("Thing creation error.");
+           console.log("Script error: " + err);
         }
-        thing.start()
-        .then (() => {
-          thing.register();
-        });
       </pre>
 
-      <pre class="example highlight" title="Create a new exposed Thing from a TD URI">
-        let thingDescription = '{
-          "name": "mySensor",
-          "url": "http://myservice.org/mySensor/description"
-        }';
-        WoT.fetch("http://myservice.org/mySensor/description")
-        .then(td => {
-          let e_thing = WoT.produce(td);
-          // properties, actions and events have been added based on the TD
-          // now add the requests handlers
-          try {
-            e_thing.setPropertyReadHandler(propertyName => {
-              console.log("Handling read request for " + propertyName);
-              return new Promise((resolve, reject) => {
-              if (this.hasOwnProperty(propertyName))
-                resolve(this[propertyName]);
-              else reject('Property not found');
-            });
-            e_thing.start();
-          } catch(error) {
-            console.log("Error");
-          }
+      <pre class="example highlight" title="Create a new exposed Thing from a Thing Description">
+        let thingDescription = '{ "@context": [ "https://w3c.github.io/wot/w3c-wot-td-context.jsonld", "https://w3c.github.io/wot/w3c-wot-common-context.jsonld" ], "@type": [ "Thing", "Sensor" ], "name": "mySensor", "geo:location": "testspace", "interaction": [ { "@type": [ "Property", "Temperature" ], "name": "prop1", "schema": { "type": "number" }, "saref:TemperatureUnit": "degree_Celsius" } ] }';
+        try {
+          // note that produce() fails if thingDescription contains error
+          let thing = WoT.produce(thingDescription);
+          // Interactions were added from TD
+          // add server functionality
+          thing.setPropertyReadHandler( (propertyName) => {
+            console.log("Handling read request for " + propertyName);
+            return new Promise((resolve, reject) => {
+              resolve(Math.random(100));
```

Comment:
You need the test of `hasOwnProperty`.

Reply:
The original example was better IMHO.

Reply:
Where does `hasOwnProperty` come from? So far it is not defined.

Why go the way of a single handler with your own multiplexer in the first place, when you rely on the Thing again to manage state and presence?

When the handler is given for a concrete Interaction, no check is needed (this case).

When the handler is for all Interactions, then the handler must be able to check is it holds the state. Experience on this is available from LWM2M implementation.

Reply:
From ECMAScript.

Reply:
Originally we had the names provided, but we discussed in the scripting call with Daniel and turned out that in a real script context we don't need them. Could you provide examples where you can't figure out the context without the names provided?

Reply:
If think you are discussing the generic handlers here, while the example shows a specific handler for `prop1`. As discussed in the call, I would not even model the `propertyName` argument for this specific handler.

I guess what you want is an example for a generic handler that shows the multiplexing based on the passed name?

Reply:
No, the original example was about a generic handler that performs the same things for all property reads. Given the property name, it fetches and then returns the property value. Hence you don't need to write (or bind) the same function with each and every property supported (suppose there are 20 properties). It just saves space. But this was not meant for providing a multiplexed implementation for a given interaction type. That was not supported.

Reply:
I suggest we separate these examples, i.e.have one as generic handler without the need to refer to a certain property (the original example), the other for in-handler dispatching of property handling, and another for specific handler that is not using the property name (since it was specifically bound to a property).

Reply:
So at least: yes, you wanted an example for a generic handler.

If there are 20 Properties, I don't see the point of returning the same result for all of them. To individualize, you need the multiplexing I am talking about (based on the propertyName passed to the handler).

I would add a generic handler for two different Actions. Having a generic handler for a single Property does not drive the point.

Reply:
It's not returning the same result! Read the code you have removed.
Internally, there are slots for all properties. The handler gets the name. So the implementation can fetch the value of the requested one. That is not multiplexing. It's parametrization. Multiplexing is another example we did not have so far.